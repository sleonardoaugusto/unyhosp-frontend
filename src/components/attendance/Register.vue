<template>
  <div class="d-flex flex-wrap justify-content-center align-items-center flex-column">
    <div class="col-md-6 mb-3">
      <div class="row">
        <h2 class="type-h4 type-light-green">Registrar atendimento</h2>
      </div>
    </div>
    <div class="col-md-6">
      <div class="row d-flex flex-column mb-3">
        <label for="PacientName">Paciente:</label>
        <input type="text" id="PacientName" v-model="inputValue" @input="searchPacients">
        <ul class="live-search__dropdown">
          <li class="live-search__item" v-for="(p, i) in pacients" :key="i" @click="setPacient(p)">{{ p.name }}</li>
        </ul>
      </div>
    </div>
  </div>
</template>

<script>
import PacientService from '@/services/PacientService'

export default {
  props: ['hospitalId'],
  name: 'Register',
  data: () => ({
    inputValue: '',
    pacients: []
  }),
  methods: {
    async searchPacients() {
      await PacientService.getPacients()
        .then((res) => {
          this.pacients = res.data
        })
    },
    setPacient(value) {
      this.inputValue = value.name
    }
  }
}
</script>

<style scoped>

</style>