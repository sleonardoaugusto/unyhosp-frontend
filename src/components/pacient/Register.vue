<template>
  <div class="d-flex flex-wrap justify-content-center align-items-center flex-column">
    <div class="col-md-6 mb-3">
      <div class="row">
        <h2 class="type-h4 type-blue-smoke">Cadastrar Paciente</h2>
      </div>
    </div>
    <div class="col-md-6">
      <div class="row d-flex flex-column mb-3">
        <label for="PacientName">Paciente:</label>
        <input type="text" id="PacientName" v-model.trim="data.name">
      </div>
      <div class="row d-flex flex-column mb-3">
        <label for="DocumentId">CPF:</label>
        <the-mask type="text" id="DocumentId" mask="###.###.###-##" v-model.trim="data.document_id"/>
      </div>
      <div class="row d-flex flex-column mb-3">
        <label for="Email">Email:</label>
        <input type="text" id="Email" v-model.trim="$v.data.email.$model">
        <span v-if="!$v.data.email.email" class="error">Deu ruim</span>
      </div>
      <div class="row d-flex flex-column mb-3">
        <label for="DateOfBirth">Data de Nascimento:</label>
        <input type="tel" id="DateOfBirth" v-mask="'##/##/####'" v-model.trim="data.date_of_birth">
      </div>
    </div>
    <div class="col-md-6">
      <div class="row">
        <button class="button button-green" @click="registerPacient">CADASTRAR</button>
      </div>
    </div>
  </div>
</template>

<script>
import PacientService from '@/services/PacientService'
import { required, email } from 'vuelidate/lib/validators'

export default {
  name: 'Register',
  data: () => ({
    data: {
      name: '',
      document_id: '',
      email: '',
      date_of_birth: ''
    }
  }),
  validations: {
    data: {
      email: { required, email }
    }
  },
  methods: {
    registerPacient() {
      PacientService.post(this.data)
    }
  }
}
</script>

<style scoped>

</style>