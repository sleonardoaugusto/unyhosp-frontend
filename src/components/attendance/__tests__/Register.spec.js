import { shallowMount } from '@vue/test-utils'
import Register from '@/components/attendance/Register'

describe('Register.vue', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(Register)
  })

  it('component is rendering properly', () => {
    wrapper.setData({ pacients: [{ name: 'Paul Green' }] })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should get pacients when interact with pacient field', () => {
    const getPacients = jest.fn()
    wrapper.setMethods({ getPacients })
    wrapper.find('input').trigger('input')
    expect(getPacients).toHaveBeenCalled()
  })

  it('call setData method when interact with live search items', () => {
    const setPacient = jest.fn()
    wrapper.setMethods({ setPacient })
    wrapper.setData({ pacients: [{ name: 'Mariazinha' }] })
    wrapper.find('.live-search__item').trigger('click')
    expect(setPacient).toHaveBeenCalled()
  })

  it('setPacient method must set a new inputValue', () => {
    const data = { name: 'Maria Joaquina Almeida' }
    wrapper.vm.setPacient(data)
    expect(wrapper.vm.inputValue).toBe(data.name)
  })
})