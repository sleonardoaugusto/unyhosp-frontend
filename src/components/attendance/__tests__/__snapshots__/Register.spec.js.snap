// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register.vue component is rendering properly 1`] = `
<div class="d-flex flex-wrap justify-content-center align-items-center flex-column">
  <div class="col-md-6 mb-3">
    <div class="row">
      <h2 class="type-h4 type-blue-smoke">Registrar atendimento</h2>
    </div>
  </div>
  <div class="col-md-6">
    <div class="row d-flex flex-column mb-3"><label for="PacientName">Paciente:</label> <input type="text" id="PacientName">
      <ul class="live-search__dropdown" style="display: none;">
        <li class="live-search__item">Paul Green</li>
        <router-link-stub to="[object Object]" tag="li" event="click" class="live-search__item type-blue-smoke type-h8">Novo
          Paciente +
        </router-link-stub>
      </ul>
    </div>
  </div>
</div>
`;
